lib/megaLogger.js

```diff
@@ -252,55 +232,48 @@
      * Send the passed `arguments` to the specific `level` logger
      *
      * @param level {number}
-     * @param arguments {*}
+     * @param args {*}
      * @private
      */
-    MegaLogger.prototype._log = function(level, arguments) {
-        var self = this;
-
-        var levelName = MegaLogger._intToLevel(level);
-        var clr = self.options.levelColors[levelName];
-        var logStyle = "color: " + "white" + "; background-color: " +  clr + "; padding-left: 1px; padding-right: 1px;";
+    MegaLogger.prototype._log = function(level, args) {
+        var options = this.options;
 
-
-        var args = [
-            (self.options.colorsEnabled ? "%c" : "") + self.options.dateFormatter(new Date()) + " - " + self._getLoggerPath() + " - " + levelName,
-            (self.options.colorsEnabled ? logStyle : "")
-        ];
-
-        args = args.concat(arguments);
-        if(self.options.dereferenceObjects) {
-            args = JSON.parse(JSON.stringify(args));
+        // check min log level
+        if (level < options.minLogLevel()) {
+            return;
```

Comment:
Here too many stuff was being unnecessarily performed **if** the level isn't high enough or logging is disabled. So, we will be now checking the level first, and i've also moved all that within the `isEnabled` block.